### ISSUE TITLE
Fails to parse requirements correctly

### ISSUE THREAD

**Issue by jdidion** [common-workflow-language/cwljava#31]
When I parse this workflow, the resulting `CommandLineToolImpl` contains two copies of the `InlineJavascriptRequirement`.

```yaml
cwlVersion: v1.2

class: CommandLineTool

requirements:
  - class: InlineJavascriptRequirement
  - class: InitialWorkDirRequirement
    listing:
      - entryname: emptyWritableDir
        entry: "$({class: 'Directory', listing: []})"
        writable: true

hints:
  - class: DockerRequirement
    dockerPull: alpine

inputs: []
outputs:
  out:
    type: Directory
    outputBinding:
      glob: emptyWritableDir
arguments: [touch, emptyWritableDir/blurg]
```

**Comment by mr-c** [comment]
Interesting! The Python parser does not have this bug..

**Comment by mr-c** [comment]
Okay, I can confirm the bug on the Java side

``` java
package org.w3id.cwl.cwl1_2.utils;

import java.nio.file.FileSystems;
import org.junit.Assert;
import org.junit.Test;
import org.w3id.cwl.cwl1_2.CWLVersion;
import org.w3id.cwl.cwl1_2.InlineJavascriptRequirement;
import org.w3id.cwl.cwl1_2.Process;

public class ExamplesTest {
  Process doc =
      (Process)
          RootLoader.loadDocument(
              FileSystems.getDefault().getPath("/home/michael/cwljava/java-31.cwl"));

  @Test
  public void className() {
    Assert.assertEquals(doc.getClass().getSimpleName(), "CommandLineToolImpl");
  }

  @Test
  public void version() {
    java.util.Optional<CWLVersion> version = doc.getCwlVersion();
    Assert.assertTrue(version.isPresent());
    Assert.assertEquals(CWLVersion.V1_2, version.get());
  }

  @Test
  public void hints() {
    java.util.Optional<java.util.List<Object>> hints = doc.getHints();
    Assert.assertTrue(hints.isPresent());
    java.util.List<Object> hintList = hints.get();
    Assert.assertEquals(hintList.size(), 1);
  }

  @Test
  public void reqs() {
    java.util.Optional<java.util.List<Object>> reqs = doc.getRequirements();
    Assert.assertTrue(reqs.isPresent());
    java.util.List<Object> reqList = reqs.get();
    Assert.assertEquals(reqList.size(), 2);
    InlineJavascriptRequirement reqOne = (InlineJavascriptRequirement) reqList.get(0);
    Assert.assertEquals(reqOne.getClass().getSimpleName(), "InlineJavascriptRequirementImpl");
    Assert.assertNotEquals(
        reqList.get(1).getClass().getSimpleName(), "InlineJavascriptRequirementImpl");
  }
}
```

**Comment by mr-c** [comment]
The reason is that
1) the Java loaders do not verify the value of the `class` field, and since `InlineJavascriptRequirement` has no mandatory fields, it erroneously accepts the `InitialWorkDirRequirement` map

Compare https://github.com/common-workflow-lab/cwljava/blob/952d107fb26d77a94d043f4668a1eeee83952ce0/src/main/java/org/w3id/cwl/cwl1_2/InlineJavascriptRequirementImpl.java#L83 to https://github.com/common-workflow-language/cwl-utils/blob/61df4b08a368098951864e2cc2adaf682a8e12f8/cwl_utils/parser_v1_2.py#L3592

Python implementation is at https://github.com/common-workflow-language/schema_salad/blob/5e8fe097babff522cf30cd2c2f49f371a6162fa8/schema_salad/python_codegen.py#L196


2) The Java loaders do not reject extra fields; here the extra `listing` field should have failed parsing as an `InlineJavascriptRequirement`

Example:
https://github.com/common-workflow-language/cwl-utils/blob/61df4b08a368098951864e2cc2adaf682a8e12f8/cwl_utils/parser_v1_2.py#L3623

**Comment by mr-c** [comment]
Fixes are inbound:
https://github.com/common-workflow-lab/cwljava/pull/32
https://github.com/common-workflow-language/cwl-v1.2/pull/62
https://github.com/common-workflow-language/schema_salad/pull/362

**Comment by mr-c** [comment]
Fixed in #32 and test added in #34
